### DISCUSSION TITLE
Feature Request: Auto Record OUTGOING Sessions

### DISCUSSION THREAD

**Discussion by midshoreit** [rustdesk/rustdesk#5453] [Q&A] [answered]
I've used RustDesk for a LONG time now, and the one feature that is missing is the ability to have recordings made automatically on OUTGOING sessions. I know it exists for INCOMING sessions, but that does ME no good as the person providing tech support to the computer I am connecting to, and remembering to hit the hidden record button at the start of each session has proven impossible with everything else I have going on. I'm thinking it would be another checkbox to auto-record all outgoing sessions, which would be right there with the destination text block.

I would gladly pay to have such a feature added, but to me, it just makes logical sense for this to be an option for people like me. Thanks for all the hard work on making RustDesk amazing. I will be signing up for a pro account soon, but I work for myself and my business isn't strong enough yet to where I can pay for an annual subscription at the moment. I could do a monthly, but you aren't accepting those, which is why I have to wait.

Thanks!

**Comment by bilogic** [comment]
Yea, I suggest just duplicate the current UI for outgoing as well since some may also need to save the sessions to a different paths, might as well do it together.

**Comment by rgigithub01** [comment]
We also would like an option to automatically record outgoing sessions. It would be an mandatory features for us.

**Comment by Codeweep** [comment]
We're thinking about switching to RustDesk and this feature would be a necessity for us too. I acknowledge that it is possible for anybody to create their own fix for this issue but I don't have any experience with flutter or Rust and this feature is something that TeamViewer has been offering for years now. 
Any news on this would be great <3

**Comment by itservicescscon** [comment]
Hey Guys, we bought the Basic Pro Version mostly for the client configurator. However as the topic says we desperatly need a option for recording our outgoing RustDesk Support with our Clients.
The option for the Clients to automatic record incoming connections is implemented but outgoing is not.
We still need to press the record Button every time we would like to record!
Can this one be implemented in the future or am i missing something and its already there?

appreciate the feedback
kind regards
cscon GmbH

**Comment by rustdesk** [comment]
@21pages let's do this after web client and qos done.

**Comment by whilo** [comment]
I would also be interested in that. Ideally I would like to run a headless linux client that automatically records my screen on a server. I have started experimenting with the code base and would be willing to contribute. Some guidance for this would be appreciated (to make sure the changes can be merged).

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/pull/9711
